src/main/java/codeu/model/data/Conversation.java

```diff
@@ -55,22 +55,35 @@ public Conversation(UUID id, UUID owner, String title, Instant creation) {
   }
 
   /**
-   * Constructs a new Conversation.
+   * Constructs a new Direct Message or Group Conversation.
    *
    * @param id the ID of this Conversation
    * @param owner the ID of the User who created this Conversation
    * @param title the title of this Conversation
    * @param creation the creation time of this Conversation
-   * @param users the Users that will be able to access and chat in this conversation
+   * @param users the Users or user names that will be able to access and chat in this conversation
    * @param conversationType the type of Conversation
    */
-  public Conversation(UUID id, UUID owner, String title, Instant creation, List<User> users,
+  public Conversation(UUID id, UUID owner, String title, Instant creation, List<?> users,
```

Comment:
Just realized that using a Generics List as a parameter could be confusing when constructing Conversations because other programmers wouldn't be able to tell that only type User or String is accepted. I know that there's bounded type parameters in Java (bounding the type to be a subclass or superclass of something) but the User/String classes don't really have any relation. Is there any simple way to fix this clarity issue?

Reply:
Got it, could I also add another constructor taking in List<User> which ends up getting the list of usernames and call the String constructor from that? I'm asking because a lot of code I wrote before this pull request constructs the user list using actual User objects.

Reply:
...Actually in the middle of trying to implement what I mentioned above, I realized why I tried to  use generics in the first place. 
![image](https://user-images.githubusercontent.com/11599574/43167906-94c3ec0c-8f58-11e8-85fb-287bc2cc1e44.png)
This doesn't let me use two constructors, one taking in user objects and one taking in user names because they technically still have the same method signature and List<T> is considered the same in both. I'll probably just refactor my old code to use usernames unless I can think of another solution.

Reply:
Maybe move the construction of the list of strings to a utility class that can be used in all places. 

Yeah, generics in Java are implemented by type erasure, which effectively means that at run time all the generic type parameters are stripped away (which means they can't be used to distinguish overloaded methods).

Reply:
Just did this using a ConversationHelper class, and the info about generics in Java is really useful to know!

I've implemented the suggestions as well as fixed up other parts of the code in the newest commit and pushed it to this pull request as well